csharp-mode.el

```diff
@@ -8,6 +8,7 @@
 ;; Version    : 0.11.0
 ;; Keywords   : c# languages oop mode
 ;; X-URL      : https://github.com/emacs-csharp/csharp-mode
+;; Package-Requires: ((emacs "26.1") (tree-sitter "0.12.1") (tree-sitter-indent "0.1"))
```

Comment:
Why do we need to move this? Is it for ease of installing or some other reason?

Reply:
Actually, this is up to you. Do you want `tree-sitter` and `tree-sitter-indent` to be our dependencies? I am getting the error #208 while installing it. But now we are using `with-eval-after-load` then I assumed this can be ignored. 😕

Reply:
Right, this makes sense. Let's keep it your way!

Reply:
We can either do it this way or split into a isolated package since `csharp-tree-sitter.el` isn't hardly rely on `csharp-mode.el`. Pros and cos are

#### Pros

1. Less dependencies for user who only uses `csharp-mode.el` and not `tree-sitter` support.

#### Cons

1. We will have to send in another recipe to melpa.
2. Of course, more work. :)

Reply:
After thinking a little about this, I am leaning towards keeping those requires solely in `csharp-tree-sitter`for some reasons:
- Tree sitter support is still experimental
- I don't think we are ready for it to be its own package yet
- I agree that it should get its own recipe, but maybe down the line a little
- This will not mess up anything for the *BSD users. 

The more I think about this the more I think we should try to pollute `csharp-mode` as little as possible with tree-sitter related stuff. At least until we can safely support *BSD and friends.

Reply:
Got it. I have moved the package requires back to it's file.